frontend/components.js

```diff
@@ -205,41 +228,73 @@ const TransactionHistory = connect('transactionHistory')(({transactionHistory})
   )
 )
 
+const ConfirmTransactionDialog = connect(
+  (state) => ({
+    sendAddress: state.sendAddress.fieldValue,
+    totalAmount: (parseFloat(state.sendAmount.fieldValue) + state.transactionFee / 1000000).toFixed(
+      6
+    ),
+  }),
+  actions
+)(({sendAddress, totalAmount, submitTransaction, cancelTransaction}) =>
+  h(
+    'div',
+    {class: 'overlay'},
+    h(
+      'div',
+      {class: 'box'},
+      h('p', undefined, 'Confirm, that you really want to send'),
+      h('div', undefined,
+        h('strong', undefined, `${totalAmount} ADA `),
+        'to the address'),
+      h('div', {class: 'address-iniline'}, sendAddress),
+      h('button', {class: 'positive', onClick: submitTransaction}, 'Confirm'),
+      h('button', {class: 'danger', onClick: cancelTransaction}, 'Cancel'),
+    )
+  )
+)
+
 class SendAdaClass extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      totalAmount: (
-        parseFloat(props.sendAmountFieldValue || 0) +
-        props.transactionFee / 1000000
-      ).toFixed(6),
+
+  render({
+    sendSuccess,
+    sendAddress,
+    sendAddressValidationError,
+    sendAmount,
+    sendAmountValidationError,
+    updateAddress,
+    updateAmount,
+    transactionFee,
+    confirmTransaction,
+    calculateFee,
+    balance,
+    showConfirmTransactionDialog,
+    feeRecalculating,
+    totalAmount,
+  }) {
+    const enableSubmit = sendAmount && !sendAmountValidationError &&
+      sendAddress && !sendAddressValidationError
+
+    if (parseFloat(sendAmount) + transactionFee > balance) {
+      sendAmountValidationError = 'INSUFFICIENT_FUNDS_FOR_FEE'
     }
-  }
 
-  componentWillReceiveProps(next) {
-    // don't update to invalid value while waiting for fee
-    if (next.feeCalculatedFrom === next.sendAmountFieldValue) {
-      this.setState({
-        totalAmount: (parseFloat(next.sendAmountFieldValue || 0) + next.transactionFee).toFixed(6),
-      })
+    const displayTransactionFee = (sendAmount !== '') && !feeRecalculating &&
+      transactionFee > 0 &&
+      (!sendAmountValidationError || sendAmountValidationError === 'INSUFFICIENT_FUNDS_FOR_FEE')
+
+    const validationMsg = (errorCode) => {
+      const errorCodeMap = {
+        INVALID_ADDRESS: 'Invalid address',
```

Comment:
this translations map belongs to a separate file imho. It would be nicer to have a "translations.js" file which would export the map with them. If you need to insert a value, you can put a closure in the map (eg INSUFFICIENT_FUNDS: (balance) => `Insufficient funds. Your balance is ${balance} ADA.` For consistency it would be nice to have all the translations as functions that by default take no parameters.